pkg/tests/ossm/ior_test.go

```diff
@@ -0,0 +1,322 @@
+package ossm
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+type RouteMetadata struct {
+	Uid             string            `json:"uid"`
+	Name            string            `json:"name"`
+	ResourceVersion string            `json:"resourceVersion"`
+	Labels          map[string]string `json:"labels"`
+	Annotations     map[string]string `json:"annotations"`
+}
+
+type Route struct {
+	Metadata RouteMetadata `json:"metadata"`
+}
+
+const (
+	originalHostAnnotation = "maistra.io/original-host"
+)
+
+// TestIOR tests IOR error regarding routes recreated: https://issues.redhat.com/browse/OSSM-1974. IOR will be deprecated on 2.4 and willl be removed on 3.0
+func TestIOR(t *testing.T) {
+	test.NewTest(t).Groups(test.Full).Run(func(t test.TestHelper) {
+		t.Log("This test verifies the behavior of IOR.")
+
+		meshNamespace := env.GetDefaultMeshNamespace()
+		meshName := env.GetDefaultSMCPName()
+
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+
+		host := "www.test.ocp"
+
+		createSimpleGateway := func(t test.TestHelper) {
+			t.Logf("Creating Gateway for %s host", host)
+			oc.ApplyString(t, "", generateGateway("gw", meshNamespace, host))
+		}
+
+		checkSimpleGateway := func(t test.TestHelper) {
+			t.Logf("Checking whether a Route is generated for %s", host)
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				routes := getRoutes(t, meshNamespace)
+				found := routes[0].Metadata.Annotations[originalHostAnnotation]
```

Comment:
Why get all routes and then take the first route instead of getting a specific route by name or label? This would make the test much more robust.

Reply:
The routes are managed by IOR only. I have added check to make sure there is only [one](https://github.com/maistra/maistra-test-tool/pull/375/files#diff-76a741e45a0b32014a31d2b4419dd39a56249aeeec8c494d9b82090f19af10ffR57) route here.